### PR TITLE
Re-add EA.RazorCompiler IVT

### PR DESCRIPTION
**Pull request by jjonescz**
Fixes https://github.com/dotnet/roslyn/issues/72015.

### CODE REVIEW

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj [hunk 1]
```diff
@@ -80,6 +80,7 @@
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.ExpressionCompiler.UnitTests" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.ExpressionCompiler.UnitTests" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExpressionEvaluator.ExpressionCompiler.Utilities" />
+    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler" />
```
Comment (jasonmalinowski) [inline src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj:83]: You may want to add a comment why this is here.

**Review by jasonmalinowski — APPROVED** [review]

**Comment by JoeRobich** [comment]
I merged my PR with the project load tests. Will need to merge in main and remove the ignored diagnostics. 

Lines https://github.com/dotnet/roslyn/blob/d8f4c5e29b3f988279ea48842eeba3b4c884e72a/src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs#L68-L80
can become
```c#
await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp);
```

File: src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs [hunk 2]
```diff
@@ -65,19 +65,7 @@ public NewlyCreatedProjectsFromDotNetNew(ITestOutputHelper output)
...
-                        // The type or namespace name {'csharp_blazor_project'|'App'} could not be found
-                        // (are you missing a using directive or an assembly reference?)
-                        // Bug: https://github.com/dotnet/roslyn/issues/72015
-                        "CS0246",
-                    ],
-                _ => Array.Empty<string>(),
-            };
-
-            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp);
```
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs:68]: We have a tracking issue (https://github.com/dotnet/roslyn/issues/72071) to expand the test coverage to cover more SDKs. Right now we are just validating it works with our pinned SDK, which is a start. =)

**Review by 333fred — APPROVED** [review]

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**